pkg/domain/entities/artifact.go

```diff
@@ -12,6 +12,7 @@ import (
 type ArtifactAddOptions struct {
 	Annotations  map[string]string
 	ArtifactType string
+	Append       bool
```

Comment:
should we make the append option incompatible with the artifact type option? I think so ?